server/service/queries.go

```diff
@@ -263,6 +263,17 @@ func createQueryEndpoint(ctx context.Context, request interface{}, svc fleet.Ser
 	return createQueryResponse{query, nil}, nil
 }
 
+func (svc *Service) getTeamName(ctx context.Context, teamID *uint) (*string, error) {
+	if teamID == nil {
+		return nil, nil
+	}
+	team, err := svc.ds.Team(ctx, *teamID)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "fetching team name")
+	}
+	return &team.Name, nil
+}
+
```

Comment:
Rather than implementing a new method just to get the team name, you can utilize the `EnterpriseOverrides` pattern to conditionally retrieve the team name if we're on a licensed tier, e.g.

```go
var teamName *string
if query.TeamID != nil {
	team, err := svc.EnterpriseOverrides.TeamByIDOrName(ctx, query.TeamID, nil)
	if err != nil {
		return nil, err
	}
	teamName = &team.Name
}
```